src/goto-symex/goto_symex_state.cpp

```diff
@@ -241,9 +217,10 @@ void goto_symex_statet::assignment(
   // the type might need renaming
   rename(lhs.type(), l1_identifier, ns);
   lhs.update_type();
-  assert_l1_renaming(lhs);
+  const validation_modet vm = global_validation_mode;
+  DATA_CHECK(!check_renaming_l1(lhs), "lhs renaming failed on l1");
```

Comment:
What I'd do here is something along the lines of `!validation_enabled || !check_renaming_l1(lhs)` to avoid changing the `validation_modet` enum.

Reply:
I reverted the new validation mode. I'm not sure if you're suggesting not to use `DATA_CHECK` at all and just call the regular invariants, or not. Have a look at the new version.